arviz/tests/test_data.py

```diff
@@ -617,20 +639,46 @@ def test_inference_data_input_types3(self, paths, observed_data_paths):
             assert hasattr(inference_data, "posterior_predictive")
             assert hasattr(inference_data, "observed_data")
 
-    def test_inference_data_input_types4(self, paths, observed_data_paths):
+    def test_inference_data_input_types4(self, paths):
         """Check input types (change, see earlier)
 
             coords --> one to many + one to one (non-default dim)
             dims --> one to many + one to one
         """
+
+        path = paths["combined_no_warmup"]
+        for path in [path, path[0]]:
+            inference_data = self.get_inference_data(
+                posterior=path,
+                posterior_predictive=path,
+                prior=path,
+                prior_predictive=path,
+                observed_data=None,
+                observed_data_var=None,
+                coords={"rand": np.arange(3)},
+                dims={"x": ["rand"]},
+            )
+            assert hasattr(inference_data, "posterior")
```

Comment:
Same comment as above here. We could make a helper method to repeat code duplication